app/src/main/java/org/wikipedia/auth/AccountUtil.kt

```diff
@@ -0,0 +1,160 @@
+        }
+        setPassword(result.password!!)
+        putUserIdForLanguage(result.site.languageCode(), result.userId)
+        groups = result.groups
+    }
+
+    @JvmStatic
+    val isLoggedIn: Boolean
+        get() = account() != null
+
+    @JvmStatic
+    val userName: String?
+        get() {
+            val account = account()
+            return account?.name
+        }
+
+    @JvmStatic
+    val password: String?
+        get() {
+            val account = account()
+            return if (account == null) null else accountManager().getPassword(account)
+        }
+
+    @JvmStatic
+    fun getUserIdForLanguage(code: String): Int {
+        val map = userIds
+        val id = map[code]
+        return id ?: 0
+    }
+
+    @JvmStatic
+    fun putUserIdForLanguage(code: String, id: Int) {
+        val ids: MutableMap<String, Int> = HashMap()
+        ids.putAll(userIds)
+        ids[code] = id
+        userIds = ids
+    }
+
+    @JvmStatic
+    var groups: Set<String>
+        get() {
+            val account = account() ?: return emptySet()
+            val setStr = accountManager().getUserData(account, WikipediaApp.getInstance().getString(R.string.preference_key_login_groups))
+            return if (TextUtils.isEmpty(setStr)) emptySet() else GsonUnmarshaller.unmarshal(object : TypeToken<Set<String>>() {}, setStr)
+        }
+        set(groups) {
+            val account = account() ?: return
+            accountManager().setUserData(account,
+                    WikipediaApp.getInstance().getString(R.string.preference_key_login_groups),
+                    GsonMarshaller.marshal(groups))
+        }
+
+    @JvmStatic
+    fun isMemberOf(groups: Set<String?>): Boolean {
+        return groups.isNotEmpty() && !Collections.disjoint(groups, AccountUtil.groups)
+    }
+
+    @JvmStatic
+    fun removeAccount() {
+        val account = account()
+        if (account != null) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1) {
+                accountManager().removeAccountExplicitly(account)
+            } else {
+                accountManager().removeAccount(account, null, null)
+            }
+        }
+    }
+
+    @JvmStatic
+    fun supported(account: Account): Boolean {
+        return account == account()
+    }
+
+    @JvmStatic
+    fun account(): Account? {
+        try {
+            val accounts = accountManager().getAccountsByType(accountType())
+            if (accounts.isNotEmpty()) {
+                return accounts[0]
+            }
+        } catch (e: SecurityException) {
+            logRemoteErrorIfProd(e)
+        }
+        return null
+    }
+
+    @JvmStatic
+    fun accountType(): String {
+        return WikipediaApp.getInstance().getString(R.string.account_type)
+    }
+
+    private fun createAccount(userName: String, password: String): Boolean {
+        var account = account()
+        if (account == null || TextUtils.isEmpty(account.name) || account.name != userName) {
```

Comment:
The `TextUtils.isEmpty()` can be replaced by `account.name..isNullOrEmpty()`

Reply:
correct!